### PR TITLE
Add web server for status and config.

### PR DESCRIPTION
**Pull request by dpkingston**
Minor refactor as well to eliminate code duplication in the main loops.

### CODE REVIEW

**Comment by dpkingston** [comment]
Its not done yet.  At a minimum I want to add parameters to the config logic.

File: config_syntax.md [hunk 1]
```diff
@@ -174,15 +174,74 @@ _Required:_ no
...
+**The port to listen on for new web connections
+
+_Type:_ Integer
+
+_Required:_ no
+
+_Default:_ 8080
+
+### use_https
+**The port to listen on for new web connections
```
Comment (bseidenberg) [inline config_syntax.md]: Fix copy/paste

File: config_syntax.md [hunk 2]
```diff
@@ -174,15 +174,74 @@ _Required:_ no
...
+**If using HTTPS, this specifies where the private key file is located
+
+_Type:_ String
+
+_Required:_ no
+
+_Default:_ ./key.pem
+
+### https_key_file
+**If using HTTPS, this specifies where the certificate file is located
```
Comment (bseidenberg) [inline config_syntax.md]: Copy paste, update

File: config_syntax.md [hunk 3]
```diff
@@ -174,15 +174,74 @@ _Required:_ no
...
+
+_Type:_ String
+
+_Required:_ no
+
+_Default:_ ./cert.pem
+
+The webserver is provided principally to allow polling of the winlink_monitor for its
```
Comment (bseidenberg) [inline config_syntax.md]: Bring up above the options for it?

Comment (bseidenberg) [reply config_syntax.md]: And add docs on the paths (basically the help)?

File: config_syntax.md (hunk 1 above)
Comment (dpkingston) [reply config_syntax.md]: Fixed.

File: config_syntax.md (hunk 2 above)
Comment (dpkingston) [reply config_syntax.md]: Fixed.

File: config_syntax.md (hunk 3 above)
Comment (dpkingston) [reply config_syntax.md]: Done.

**Review by bseidenberg — APPROVED** [review]

**Comment by dpkingston** [comment]
What is the process for updating the canary system?
I think we need to commit the updates to the config-uhf.json, and then sync
the rest of the branch with main.

-Doug-

**Comment by dpkingston** [comment]
http://winlink-canary.msv.hamwan.net:8080/status.html and
http://winlink-canary.msv.hamwan.net:8081/status.html are live! I think our
work here is nearly done. Just need to get links to these embedded on a
webpage on the Seattle ACS site. A huge thank you to Ben for pulling this
together.

Status is also available as json (/status.json).

It seems that the act of using the gateways periodically has actually
improved their reliability.  I suspect that long periods of no traffic
causes problems.  Time will tell, but our testing over the past couple of
weeks seems to suggest that strongly. We are currently polling each gateway
roughly hourly.  We can adjust this later.

-Doug-

**Comment by dpkingston** [comment]
This is a huge milestone! Kudos to all involved. Can't wait to see how it
plays out now that it's in production.

73,

Tim

**Pull request merged**